qiskit/providers/aer/quantum_info/states/aer_statevector.py

```diff
@@ -0,0 +1,204 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017, 2019, 2020, 2021, 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+Statevector quantum state class.
+"""
+import copy
+import numpy as np
+
+from qiskit.quantum_info.states import Statevector
+from qiskit.circuit import QuantumCircuit, Instruction
+from qiskit.providers.aer.quantum_info.states.aer_state import AerState
+
+
+class AerStatevector(Statevector):
+    """AerStatevector class"""
+
+    def __init__(self, data, **configs):
+        """Initialize a statevector object.
+
+        Args:
+            data (np.array or list or Statevector or QuantumCircuit or qiskit.circuit.Instruction):
+                Data from which the statevector can be constructed. This can be either a complex
+                vector, another statevector or a ``QuantumCircuit`` or ``Instruction``
+                (``Operator`` is not supportted in the current implementation).  If the data is
+                a circuit or instruction, the statevector is constructed by assuming that all
+                qubits are initialized to the zero state.
+            configs (kwargs): configurations of ```AerState`. All the keys are handeld as string
+
+        Raises:
+            QiskitError: if input data is not valid.
+
+        """
+        if '_aer_state' in configs:
+            self._aer_state = configs.pop('_aer_state')
+        if isinstance(data, (QuantumCircuit, Instruction)):
+            data, aer_state = AerStatevector._from_instruction(data, None, **configs)
+            self._aer_state = aer_state
+        elif isinstance(data, np.ndarray):
+            data, aer_state = AerStatevector._from_ndarray(data, **configs)
+            self._aer_state = aer_state
+        else:
+            self._aer_state = None
+        super().__init__(data)
+        self._result = None
+        self._configs = configs
+
+    def aer(self):
+        return self._aer_state is not None
```

Comment:
What does this do? This is a public method it should have some docstring explaining the purpose of it. But I'm not clear on the purpose.

Reply:
Let me keep it internal. `AerStatevector` has two modes to manage statevector: use of `AerState` and use of super class (`ndarray` of `q.qi.Statevector`). `aer()` returns `True` if `AerStatevector` is in the former state, otherwise, returns `False.`